extensions/interactions/PencilCodeEditor/PencilCodeEditor.js

```diff
@@ -30,11 +30,12 @@ oppia.directive('oppiaInteractivePencilCodeEditor', [
       templateUrl: 'interaction/PencilCodeEditor',
       controller: [
         '$scope', '$attrs', '$element', '$timeout', 'focusService',
-        'pencilCodeEditorRulesService',
+        'pencilCodeEditorRulesService', '$modal',
         function($scope, $attrs, $element, $timeout, focusService,
-            pencilCodeEditorRulesService) {
+            pencilCodeEditorRulesService, $modal) {
           $scope.initialCode = oppiaHtmlEscaper.escapedJsonToObj(
             $attrs.initialCodeWithValue);
+          $scope.trackResetCode = '';
```

Comment:
trackResetCode sounds like the name for a function. Perhaps something like $scope.codeBeforeReset.